src/quagga.js

```diff
@@ -1,268 +1,40 @@
-    }
-}
-
-function hasCodeResult(result) {
-    return result && (result.barcodes ?
-        result.barcodes.some(barcode => barcode.codeResult) :
-        result.codeResult);
-}
-
-function publishResult(result, imageData) {
-    let resultToPublish = result;
-
-    if (result && _context.onUIThread) {
-        transformResult(result);
-        addResult(result, imageData);
-        resultToPublish = result.barcodes || result;
-    }
-
-    Events.publish('processed', resultToPublish);
-    if (hasCodeResult(result)) {
-        Events.publish('detected', resultToPublish);
-    }
-}
-
-function locateAndDecode() {
-    const boxes = getBoundingBoxes();
-    if (boxes) {
-        const decodeResult = _context.decoder.decodeFromBoundingBoxes(boxes) || {};
-        decodeResult.boxes = boxes;
-        publishResult(decodeResult, _context.inputImageWrapper.data);
-    } else {
-        const imageResult = _context.decoder.decodeFromImage(_context.inputImageWrapper);
-        if (imageResult) {
-            publishResult(imageResult, _context.inputImageWrapper.data);
-        } else {
-            publishResult({ codeResult: { code: null } });
-        }
-    }
-}
-
-function update() {
-    if (_context.onUIThread) {
-        const workersUpdated = QWorkers.updateWorkers(_context.framegrabber);
-        if (!workersUpdated) {
-            _context.framegrabber.attachData(_context.inputImageWrapper.data);
-            if (_context.framegrabber.grab()) {
-                if (!workersUpdated) {
-                    locateAndDecode();
-                }
-            }
-        }
-    } else {
-        _context.framegrabber.attachData(_context.inputImageWrapper.data);
-        _context. _framegrabber.grab();
-        locateAndDecode();
-    }
-}
-
-function startContinuousUpdate() {
-    var next = null,
-        delay = 1000 / (_context.config.frequency || 60);
-
-    _context.stopped = false;
-    (function frame(timestamp) {
-        next = next || timestamp;
-        if (!_context.stopped) {
-            if (timestamp >= next) {
-                next += delay;
-                update();
-            }
-            window.requestAnimFrame(frame);
-        }
-    }(performance.now()));
-}
-
-function start() {
-    if (_context.onUIThread && _context.config.inputStream.type === 'LiveStream') {
-        startContinuousUpdate();
-    } else {
-        update();
-    }
-}
-
-function setReaders(readers) {
-    if (_context.decoder) {
-        _context.decoder.setReaders(readers);
-    }
-    QWorkers.setReaders(readers);
-}
-
-function registerReader(name, reader) {
-    // load it to the module
-    BarcodeDecoder.registerReader(name, reader);
-    // then make sure any running instances of decoder and workers know about it
-    if (_context.decoder) {
-        _context.decoder.registerReader(name, reader);
-    }
-    QWorkers.registerReader(name, reader);
-}
+const instance = new Quagga();
+const _context = instance.context;
```

Comment:
I'd prefer to just access everything from `instance` itself - keeps it easier to reason about where the data is coming from.

Reply:
If that's the only suggestion, then I feel pretty good about the change itself being good.  I'll definitely take that into consideration.